src/userEvent/touchEvents.ts

```diff
@@ -0,0 +1,115 @@
+import { ReactTestInstance } from 'react-test-renderer';
+import act from '../act';
+import { getHostParent } from '../helpers/component-tree';
+import { filterNodeByType } from '../helpers/filterNodeByType';
+import { isHostElementPointerEventEnabled } from '../helpers/isHostElementPointerEventEnabled';
+import { getHostComponentNames } from '../helpers/host-component-names';
+import { jestFakeTimersAreEnabled } from '../helpers/timers';
+import { DEFAULT_MIN_PRESS_DURATION } from './constants';
+import { warnAboutRealTimers } from './utils/warnAboutRealTimers';
+
+const defaultPressEvent = {
+  persist: jest.fn(),
+  nativeEvent: {
+    timestamp: new Date().getTime(),
+  },
+  currentTarget: { measure: jest.fn() },
+};
+
+export type PressOptions = {
+  pressDuration: number;
+};
+
+export const basePress = async (
+  element: ReactTestInstance,
+  options: PressOptions = { pressDuration: 0 }
+): Promise<void> => {
+  // Text and TextInput components are mocked in React Native preset so the mock
+  // doesn't implement the pressability class
+  // Thus we need to call the props directly on the host component
+  const isEnabledHostText =
+    filterNodeByType(element, getHostComponentNames().text) &&
+    !element.props.disabled;
+  const isEnabledTextInput =
+    filterNodeByType(element, getHostComponentNames().textInput) &&
+    element.props.editable !== false;
+
+  if (isEnabledHostText || isEnabledTextInput) {
```

Comment:
Seems like we should apply the delay here as well. Wdyt?

Reply:
Yes you're right, I added a delay before the press and also now wait before the press out